repos/fdbt-site/src/pages/api/selectSalesOfferPackage.ts

```diff
@@ -1,34 +1,67 @@
-import isArray from 'lodash/isArray';
 import { NextApiResponse } from 'next';
+import { MULTIPLE_PRODUCT_ATTRIBUTE, SALES_OFFER_PACKAGES_ATTRIBUTE } from '../../constants/attributes';
 import {
-    SalesOfferPackage,
-    NextApiRequestWithSession,
     ErrorInfo,
+    NextApiRequestWithSession,
     ProductWithSalesOfferPackages,
+    SalesOfferPackage,
     SelectSalesOfferPackageWithError,
 } from '../../interfaces';
-import { redirectTo, redirectToError } from './apiUtils';
-import { SALES_OFFER_PACKAGES_ATTRIBUTE, MULTIPLE_PRODUCT_ATTRIBUTE } from '../../constants/attributes';
 import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
-import { removeAllWhiteSpace } from './apiUtils/validator';
+import { redirectTo, redirectToError } from './apiUtils';
+import { checkPriceIsValid, removeAllWhiteSpace, removeExcessWhiteSpace } from './apiUtils/validator';
 
 interface SanitisedBodyAndErrors {
-    sanitisedBody: { [key: string]: string[] };
+    sanitisedBody: { [key: string]: SalesOfferPackage[] };
     errors: ErrorInfo[];
 }
 
+const productPrefix = 'product-';
+const pricePrefix = 'price-';
+
 export const sanitiseReqBody = (req: NextApiRequestWithSession): SanitisedBodyAndErrors => {
-    const sanitisedBody: { [key: string]: string[] } = {};
-    const errors: ErrorInfo[] = [];
+    const sanitisedBody: { [key: string]: SalesOfferPackage[] } = {};
 
+    const errors: ErrorInfo[] = [];
     Object.entries(req.body).forEach(item => {
-        if (item[1] && isArray(item[1])) {
-            sanitisedBody[item[0]] = (item[1] as string[]).filter(a => a !== '');
+        const [key, value]: [string, unknown] = item;
+
+        if (key.startsWith(productPrefix)) {
+            if (value && Array.isArray(value)) {
+                sanitisedBody[key.substring(productPrefix.length)] = value.filter(a => a).map(it => JSON.parse(it));
+            } else {
+                errors.push({
+                    errorMessage: 'Choose at least one sales offer package from the options',
+                    id: `${removeAllWhiteSpace(key)}-checkbox-0`,
+                });
+            }
+        } else if (key.startsWith(pricePrefix) && typeof value === 'string') {
```

Comment:
should these be swapped around?  typeof value === 'string' &&  key.startsWith(pricePrefix)

Reply:
Never mind, key and value are different things.